### ISSUE TITLE
LspHover issue with Conceal

### ISSUE THREAD

**Issue by habamax** [prabirshrestha/vim-lsp#719]
Currently LspHover use conceal + cchar feature to hide `\` and for escaped chars:

![image](https://user-images.githubusercontent.com/234774/74512434-4ba36e80-4f19-11ea-9f3b-92a5137a094f.png)

As a followup of using cchar it also clears `Conceal` highlighting which breaks it for other filetypes.

![image](https://user-images.githubusercontent.com/234774/74512690-e3a15800-4f19-11ea-8afb-92b19a455767.png)

The code in question is:

https://github.com/prabirshrestha/vim-lsp/blob/master/syntax/lsp-hover.vim#L49

**Comment by habamax** [comment]
What I suggest is to use another approach:

```
    if has('conceal')
        for l:escaped_char in ['`', '*', '_', '{', '}', '(', ')', '<', '>', '#', '+', '.', '!', '-']
            execute printf('syntax region vimLspMarkdownEscape matchgroup=Conceal start="\\\ze[%s]" end="[%s]\zs" concealends', l:escaped_char, l:escaped_char)
        endfor
    end

    hi link vimLspMarkdownEscape Pmenu
```

Which makes:

1. all those escaped chars to be highlighted as Pmenu -- thus it wouldn't have different background
2. will not break Conceal highlight group defined by colorscemes

![image](https://user-images.githubusercontent.com/234774/74512943-6cb88f00-4f1a-11ea-93c2-7d2db7820597.png)

**Comment by habamax** [comment]
@prabirshrestha, @mattn should I create PR?

**Comment by clason** [comment]
You can disable this altogether by registering the server with `'config': {  'hover_conceal': 0 }` .

**Comment by habamax** [comment]
> 
> 
> You can disable this altogether by registering the server with `'config': { 'hover_conceal': 0 }` .

I do not see it in the code though...

`s:cleanup_markdown` is called in any way

**Comment by clason** [comment]
Nevertheless, that's what I'm using ;)

See https://github.com/prabirshrestha/vim-lsp/blob/52539a54ae338993460205b4528d34caf175582e/autoload/lsp/ui/vim/output.vim#L327
and https://github.com/prabirshrestha/vim-lsp/blob/fabb572f7e85baaeeb57d8321ad3a901a1c5cf5a/doc/vim-lsp.txt#L801

**Comment by habamax** [comment]
> Nevertheless, that's what I'm using ;)

Again, `Conceal` highlight group is explicitly cleared.

**Comment by clason** [comment]
Oh, you're right, I misread the issue. I'll delete my comments to avoid cluttering the thread. Sorry for the noise!

**Comment by thomasfaingnaert** [comment]
Closed by #720

**Comment by Shatur95** [comment]
I sometimes have something like the following in hover message (extra "help" text with ```):
```vim
```help
description
\```
```

Is it related to conceal level?
